storybookjs/storybook
# Added a vanilla React Native example app.

Issue: We need an example app for RN!  #1012 

## What I did

```
cd examples
react-native init ReactNativeVanilla
cd ReactNativeVanilla
# --> checked it ran normally

getstorybook
# --> checked it ran -- actually there was an issue #1200 I worked around

# --> updated package.json w/ file:// ref to @storybook/react-native
npm install
# --> checked it ran -- I had to revert to RN 0.44.1 because of https://github.com/facebook/react-native/issues/14209]

# --> updated package.json w/ file:// ref to @storybook/addon-storyshots
npm install
# --> added `__tests__/storyshots.js` test file
npm test # <- ensured it worked as expected.

# --> updated package.json w/ file:// ref to @storybook/addon-options
# --> Added code to use options in `storybook/addons.js` and `storybook/index.ios.js` 
#    (note had to work around https://github.com/storybooks/storybook/issues/1192 with a `setTimeout`)
npm run storybook / react-native run-ios # <-- to check the options worked.
```

## How to test

1. Checkout, `npm install`.
2. Run the storybook via (a) `npm run storybook` + `react-native run-ios`. Check the stories are displayed and the `name` option works.
3. Run `npm test`, ensure 3 snapshots are recorded.
